### ISSUE TITLE
Roslyn fails to verify nullable attributes on lambda expressions in delegate conversions

### ISSUE THREAD

**Issue by TessenR** [dotnet/roslyn#55649]
**Version Used**: 
```
Branch main (14 Aug 2021)
Latest commit 0954614 by David Barbet:
Merge pull request #55599 from dibarbet/shutdown_jsonrpc_disconnect

Shutdown LSP server on streamjsonrpc disconnect
```

**Steps to Reproduce**:

Compile and run the following code
```cs
#nullable enable
using System;
using System.Diagnostics.CodeAnalysis;

Func<string?, string> d = ([DisallowNull] string? x) => x;
d(null).ToString();
```

**Expected Behavior**:
Warning for the assignment of `d` since the assigned function doesn't accept null values which are legal as inputs of the delegate

**Actual Behavior**:
No warnings at all in the program above. The program crashes at runtime with a `NullReferenceException`

**Comment by jcouv** [comment]
Resolved in 17.0p5

**Comment by jcouv** [comment]
Fixed in 17.1
